packages/ai/core/embed/embed-many.test.ts

```diff
@@ -18,6 +18,68 @@ const testValues = [
   'snowy night in the mountains',
 ];
 
+describe('model.supportsParallelCalls', () => {
+  it('should not parallelize when false', async () => {
+    let i = 0;
+
+    const { embeddings } = await embedMany({
+      model: new MockEmbeddingModelV2({
+        supportsParallelCalls: false,
+        maxEmbeddingsPerCall: 1,
+        doEmbed: async ({ values }) => {
+          const index = i++;
+
+          expect(values).toEqual([testValues[index]]);
+
+          await new Promise(resolve => setTimeout(resolve, 1));
```

Comment:
Use `delay` if you really need a timeout. Ideally timeouts should be avoided in tests.

Reply:
better idea: track `start` and `end` of mock calls in array, and use promises to resolve in parallel case.

Reply:
yeah this is just a 1ms delay to ensure it runs async vs sync, I can also do start / end

Reply:
done 🙏